.github/workflows/pr-build.yml

```diff
@@ -20,18 +43,66 @@ jobs:
         uses: conda-incubator/setup-miniconda@v3
         with:
           channels: conda-forge
-          environment-file: air-quality-backend/conda/etl_environment.yml
+          environment-file: air-quality-backend/conda/linting_environment.yml
           auto-activate-base: false
-          activate-environment: etl
+          activate-environment: linting-env
          
       - name: Run lint check
         run: python -m flake8 --append-config air-quality-backend/.flake8 air-quality-backend
 
-      - name: Run unit tests
-        run: python -m pytest --cov=src --cov=scripts --cov-report=xml:air-quality-backend/tests/coverage/test_report.xml air-quality-backend/tests
-        
+  etl:
+    needs: changes
+    if: ${{ needs.changes.outputs.etl == 'true' || needs.changes.outputs.etl_tests == 'true' }}
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          channels: conda-forge
+          environment-file: air-quality-backend/conda/etl_environment.yml
+          auto-activate-base: false
+          activate-environment: etl-dev
+         
+      - name: Run etl unit tests
+        run: python -m pytest --cov=src --cov=scripts --cov-report=xml:air-quality-backend/tests/coverage/etl_test_report.xml air-quality-backend/tests/etl_tests
+
+      - name: Code Coverage Report
+        uses: orgoro/coverage@v3.1
+        with:
+            coverageFile: air-quality-backend/tests/coverage/etl_test_report.xml
+            token: ${{ secrets.GITHUB_TOKEN }}
+
+  api:
+    needs: changes
+    if: ${{ needs.changes.outputs.api == 'true' || needs.changes.outputs.api_tests == 'true' }}
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          channels: conda-forge
+          environment-file: air-quality-backend/conda/api_environment.yml
+          auto-activate-base: false
+          activate-environment: api-dev
+         
+      - name: Run api unit tests
+        run: python -m pytest --cov=src --cov=scripts --cov-report=xml:air-quality-backend/tests/coverage/api_test_report.xml air-quality-backend/tests/api_tests
+
       - name: Code Coverage Report
         uses: orgoro/coverage@v3.1
         with:
-            coverageFile: air-quality-backend/tests/coverage/test_report.xml
+            coverageFile: air-quality-backend/tests/coverage/api_test_report.xml
             token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I wonder what happens if we have two coverage reports on the PR? Maybe it shows both?

Reply:
Hmnnn not sure